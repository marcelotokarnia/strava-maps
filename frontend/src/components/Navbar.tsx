import { connect, ConnectedProps } from 'react-redux'
import React, { FC } from 'react'
import { ActivitiesActions } from 'store/actions'
import Checkbox from 'components/styleguide/Checkbox'
import { length } from 'ramda'
import { useRouter } from 'next/router'

const GITHUB_LINK = 'https://www.github.com/marcelotokarnia/strava-maps'

const mapDispatchToProps = { toggleMockApi: ActivitiesActions.toggleMockApi }

const mapStateToProps = state => ({
  mockedApi: state.activities.useMockApi,
  hasActivities: !!length(state.activities.activitiesList),
})

const buttonClassName =
  'no-underline near-white bg-animate bg-near-black hover-bg-gray inline-flex items-center ma2 tc br2 pa2 pointer bn'
const stravaLoginContent = (
  <>
    <img src="/static/icons/social/strava.png" className="dib w2 br-100" alt="Site Name" />
    <span className="f6 ml3 pr2">Strava Login</span>
  </>
)
const githubButton = (
  <a className={buttonClassName} href={GITHUB_LINK} title="GitHub">
    <svg
      className="dib h2 w2"
      fill="currentColor"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 16 16"
      fillRule="evenodd"
      clipRule="evenodd"
      strokeLinejoin="round"
      strokeMiterlimit="1.414"
    >
      <path d="M8 0C3.58 0 0 3.582 0 8c0 3.535 2.292 6.533 5.47 7.59.4.075.547-.172.547-.385 0-.19-.007-.693-.01-1.36-2.226.483-2.695-1.073-2.695-1.073-.364-.924-.89-1.17-.89-1.17-.725-.496.056-.486.056-.486.803.056 1.225.824 1.225.824.714 1.223 1.873.87 2.33.665.072-.517.278-.87.507-1.07-1.777-.2-3.644-.888-3.644-3.953 0-.873.31-1.587.823-2.147-.083-.202-.358-1.015.077-2.117 0 0 .672-.215 2.2.82.638-.178 1.323-.266 2.003-.27.68.004 1.364.092 2.003.27 1.527-1.035 2.198-.82 2.198-.82.437 1.102.163 1.915.08 2.117.513.56.823 1.274.823 2.147 0 3.073-1.87 3.75-3.653 3.947.287.246.543.735.543 1.48 0 1.07-.01 1.933-.01 2.195 0 .215.144.463.55.385C13.71 14.53 16 11.534 16 8c0-4.418-3.582-8-8-8" />
    </svg>
    <span className="f6 ml3 pr2">View Source</span>
  </a>
)

const connector = connect(mapStateToProps, mapDispatchToProps)

type NavbarProps = ConnectedProps<typeof connector>

export const Navbar: FC<NavbarProps> = ({ toggleMockApi, mockedApi, hasActivities }) => {
  const router = useRouter()
  return (
    <nav className={`${router.pathname === '/map' ? 'dn' : 'dt'} w-100 border-box pa3`}>
      <div className="dtc v-mid w-75 tr">
        {!hasActivities && (
          <>
            <Checkbox
              className="inline-flex items-center"
              text="using mocked data"
              disabledText="using real data"
              onChange={value => toggleMockApi({ useMockApi: value })}
              checked={mockedApi}
            />
            {mockedApi ? (
              <button className={buttonClassName} onClick={() => router.push('/login?code=mocked')}>
                {stravaLoginContent}
              </button>
            ) : (
              <a className={buttonClassName} href="/strava/auth">
                {stravaLoginContent}
              </a>
            )}
          </>
        )}
        {githubButton}
      </div>
    </nav>
  )
}

export default connector(Navbar)
