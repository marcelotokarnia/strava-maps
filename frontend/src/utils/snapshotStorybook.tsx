import { BrowserRouter } from 'react-router-dom'
import { dissoc } from 'ramda'
import React from 'react'
import renderer from 'react-test-renderer'

export default (stories: any) => {
  return describe(stories.default.title, () => {
    Object.keys(dissoc('default', stories)).forEach(name =>
      it(name, () => {
        const tree = renderer.create(<BrowserRouter>{stories[name]()}</BrowserRouter>).toJSON()
        expect(tree).toMatchSnapshot()
      })
    )
  })
}
