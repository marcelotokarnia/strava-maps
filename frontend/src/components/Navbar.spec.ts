// THIS FILE IS AUTOGENERATED; DO NOT EDIT MANUALLY
import * as tests from './Navbar.stories'
import snapshotStorybook from '../utils/snapshotStorybook'

snapshotStorybook(tests)
