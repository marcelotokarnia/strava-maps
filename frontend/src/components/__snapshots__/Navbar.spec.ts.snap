// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component NavbarRaw 1`] = `
<nav
  className="dt w-100 border-box pa3"
>
  <div
    className="dtc v-mid w-75 tr"
  >
    <div
      className="inline-flex items-center"
    >
      <div
        className="dib mr2 helvetica silver"
      >
        using real data
      </div>
      <div
        className="relative dib"
      >
        <input
          className="absolute z-5 w-100 h-100 o-0 pointer"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          className="relative z-4 dib w3 h2 bg-mid-gray overflow-hidden br4 v-mid bg-animate "
        >
          <div
            className="absolute right-auto left-0 w2 h2 br4 bg-silver shadow-4 t-cb bg-animate "
          />
        </div>
      </div>
    </div>
    <a
      className="no-underline near-white bg-animate bg-near-black hover-bg-gray inline-flex items-center ma2 tc br2 pa2 pointer bn"
      href="https://www.strava.com/oauth/authorize?client_id=28106&redirect_uri=https://strava-maps.herokuapp.com/login&response_type=code&scope=activity:read_all"
    >
      <img
        alt="Site Name"
        className="dib w2 br-100"
        src="strava.png"
      />
      <span
        className="f6 ml3 pr2"
      >
        Strava Login
      </span>
    </a>
    <a
      className="no-underline near-white bg-animate bg-near-black hover-bg-gray inline-flex items-center ma2 tc br2 pa2 pointer bn"
      href="https://www.github.com/marcelotokarnia/strava-maps"
      title="GitHub"
    >
      <svg
        className="dib h2 w2"
        clipRule="evenodd"
        fill="currentColor"
        fillRule="evenodd"
        strokeLinejoin="round"
        strokeMiterlimit="1.414"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0C3.58 0 0 3.582 0 8c0 3.535 2.292 6.533 5.47 7.59.4.075.547-.172.547-.385 0-.19-.007-.693-.01-1.36-2.226.483-2.695-1.073-2.695-1.073-.364-.924-.89-1.17-.89-1.17-.725-.496.056-.486.056-.486.803.056 1.225.824 1.225.824.714 1.223 1.873.87 2.33.665.072-.517.278-.87.507-1.07-1.777-.2-3.644-.888-3.644-3.953 0-.873.31-1.587.823-2.147-.083-.202-.358-1.015.077-2.117 0 0 .672-.215 2.2.82.638-.178 1.323-.266 2.003-.27.68.004 1.364.092 2.003.27 1.527-1.035 2.198-.82 2.198-.82.437 1.102.163 1.915.08 2.117.513.56.823 1.274.823 2.147 0 3.073-1.87 3.75-3.653 3.947.287.246.543.735.543 1.48 0 1.07-.01 1.933-.01 2.195 0 .215.144.463.55.385C13.71 14.53 16 11.534 16 8c0-4.418-3.582-8-8-8"
        />
      </svg>
      <span
        className="f6 ml3 pr2"
      >
        View Source
      </span>
    </a>
  </div>
</nav>
`;
